addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
@stefanpenner @rwjblue thanks for the great rundown (I've learned a ton just reading the comments above). One thing I wanted to clarify quick ... because you are doing a run.join this won't join the original (sync call like a closure action from a button click as you discussed @stefanpenner) but how does this run.join work inside an integration test if I don't wrap my click event (or change event) with ember.run? I ask because as of ember 2.3 I don't wrap my integration clicks/onchanges any longer and I'm curious what the join truly "joins up" with as autorun is disabled in testing

Thank you both in advance -learning 10x from you guys :)